### PR TITLE
minor: add encoding and decoding for policy rule name

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
### Describe your changes:

This PR will add support for encoding and decoding the policy rule name when editing the rule. This ensures that characters like "/" in the rule name are not treated as part of the route.

#
### Type of change:

- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/pages/PoliciesPage/PoliciesDetailPage/PoliciesDetailPage.tsx [hunk 1]
```diff
@@ -253,7 +254,11 @@ const PoliciesDetailPage = () => {
                       onClick={(e) => {
                         e.stopPropagation();
                         history.push(
-                          getEditPolicyRulePath(fqn, rule.name || '')
+                          getEditPolicyRulePath(
+                            fqn,
+                            // rule name is same as entity fqn so we need to encode it to pass it as a param
+                            getEncodedFqn(rule.name || '')
+                          )
```
Comment (Ashish8689) [inline openmetadata-ui/src/main/resources/ui/src/pages/PoliciesPage/PoliciesDetailPage/PoliciesDetailPage.tsx:261]: @Sachin-chaurasiya  we can encode this too in the `getEditPolicyRulePath` just like we are doing for the `FQN` ?
image

File: openmetadata-ui/src/main/resources/ui/playwright/utils/form.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  fieldName: string;
+  errorDivSelector: string;
+  fieldSelector?: string;
+  checkEmpty?: boolean;
+  checkLong?: boolean;
+}) => {
+  if (checkEmpty) {
+    // Check empty name field message
+    await page.fill(fieldSelector, 'test');
+    await page.fill(fieldSelector, '');
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/utils/form.ts:34]:
```suggestion
    await page.locator(fieldSelector).clear();
```

File: openmetadata-ui/src/main/resources/ui/playwright/utils/form.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  if (checkLong) {
+    // Check long name field message
+    await page.fill(fieldSelector, 'name'.repeat(33));
+
+    await page
+      .getByText(`${fieldName} size must be between 1 and 128`)
+      .isVisible();
+
+    await page.fill(fieldSelector, '');
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/utils/form.ts:47]: same here

File: openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/Policies.spec.ts [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+  inValidCondition: 'isOwner(',
+};
+
+const errorMessageValidation = {
+  lastPolicyCannotBeRemoved: 'At least one policy is required in a role',
+  lastRuleCannotBeRemoved: 'At least one rule is required in a policy',
+};
+
+const policyName = `Policy-test-${uuid()}`;
+const description = `This is ${policyName} description`;
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/Policies.spec.ts:43]: can we move this to const, to keep test file clean

**Comment by harshach** [comment]
@chirag-madlani @Sachin-chaurasiya are we checking everywhere where we may not have added the urlencode?

**Review by ShaileshParmar11 — APPROVED** [review]

**Pull request merged**